desktop_version/src/Editor.cpp

```diff
@@ -139,117 +138,94 @@ void editorclass::register_tool(EditorTools tool, const char* name, const char*
     tool_requires_shift[tool] = shift;
 }
 
-void editorclass::gethooks(void)
-{
-    // Scan through the script and create a hooks list based on it
-    hooklist.clear();
-    for (size_t i = 0; i < script.customscripts.size(); i++)
-    {
-        Script& script_ = script.customscripts[i];
-
-        hooklist.push_back(script_.name);
-    }
-}
-
-void editorclass::loadhookineditor(const std::string& t)
+void editorclass::load_script_in_editor(const std::string& name)
 {
-    //Find hook t in the scriptclass, then load it into the editor
-    clearscriptbuffer();
+    // Load script t into the script editor
+    clear_script_buffer();
 
-    for(size_t i = 0; i < script.customscripts.size(); i++)
+    for (size_t i = 0; i < script.customscripts.size(); i++)
     {
-        Script& script_ = script.customscripts[i];
-
-        if(script_.name == t)
+        if (script.customscripts[i].name == name)
         {
-            sb = script_.contents;
+            script_buffer = script.customscripts[i].contents;
             break;
         }
     }
 
-    if(sb.empty())
+    if (script_buffer.empty())
     {
-        //Always have one line or we'll have problems
-        sb.resize(1);
+        // Always have one line or we'll have problems
+        script_buffer.resize(1);
     }
 }
 
-void editorclass::addhooktoscript(const std::string& t)
+void editorclass::remove_script(const std::string& name)
 {
-    //Adds hook+the scriptbuffer to the end of the scriptclass
-    removehookfromscript(t);
-    Script script_;
-    script_.name = t;
-    script_.contents = sb;
-    script.customscripts.push_back(script_);
-}
-
-void editorclass::removehookfromscript(const std::string& t)
-{
-    /* Find hook t in the scriptclass, then removes it (and any other code with it)
-     * When this loop reaches the end, it wraps to SIZE_MAX; SIZE_MAX + 1 is 0 */
-    size_t i;
-    for (i = script.customscripts.size() - 1; i + 1 > 0; --i)
+    for (size_t i = 0; i < script.customscripts.size(); i++)
     {
-        if (script.customscripts[i].name == t)
+        if (script.customscripts[i].name == name)
         {
             script.customscripts.erase(script.customscripts.begin() + i);
+            break;
         }
     }
 }
 
-void editorclass::removehook(const std::string& t)
+void editorclass::create_script(const std::string& name, const std::vector<std::string> contents)
```

Comment:
Small optimization here, avoids an unnecessary copy.
```suggestion
void editorclass::create_script(const std::string& name, const std::vector<std::string>& contents)
```